src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -170,53 +166,44 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
   @Override
   public void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
     log.info(String.format("Received %d offers", offers.size()));
-
-    if (liveState.getStagingTasksSize() != 0) {
-      log.info("Declining offers because tasks are currently staging");
-      for (Offer offer : offers) {
+    if (liveState.getCurrentAcquisitionPhase().equals(AcquisitionPhase.RECONCILING_TASKS)
+        && reconciliationComplete()) {
+      correctCurrentPhase();
+    }
+    boolean acceptedOffer = false;
+    for (Offer offer : offers) {
+      if (acceptedOffer) {
         driver.declineOffer(offer.getId());
```

Comment:
Ah, thanks, somehow I missed the fact that it stops looping through the offers. Yes, more complex logic is needed here to fully resolve this case. Let's leave the issue open, and I'll add a TODO.